### PR TITLE
docs: GSoC report

### PR DESCRIPTION
**Pull request by osamamagdy**
# Description
This is a Blog Post (or a report) summarizing all my work during the GSoC period on Supply Chain Security with Jenkins-X

### CODE REVIEW

File: content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+---
+title: "GSoC 2022 Final Report: Improving Supply Chain Security"
+date: 2022-11-08
+draft: false
+weight: 100
+description: >
+  Going through my GSoC journey with Jenkins-X, this works as a documentation for our work to improve supply chain security of Jenkins X
```
Comment (babadofar) [inline content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md:7]: Jenkins X, without hyphen :)

File: content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  The work done so far was going through three main sections
+  1. Integrating with Tekton Chains to sign TaskRuns and PipelineRuns
+  2. Software Bill of Materials (SBOM)
+  3. Signing Jenkins-X artifacts
+
+### Integrating with Tekton Chains to sign TaskRuns and PipelineRuns
+
+  ***Description:***
+
+    As Jenkins X uses tekton as its pipeline execution engine,` TaskRun`s and ` PipelineRun`s are considered the key components of Jenkins X pipeline `activities ` and  `steps `
```
Comment (babadofar) [inline content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md:29]: Something funky with the formatting, just remove the plural s from TaskRun and PipelineRUn

File: content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  Going through my GSoC journey with Jenkins-X, this works as a documentation for our work to improve supply chain security of Jenkins X
+categories: [blog]
+keywords: [Community, GSoC, 2022, supply chain security, Report]
+slug: "gsoc-2022"
+aliases: []
+author: Osama Magdy
+---
+## Project Description
+
+  Due to the massive increase in open source projects usage in different applications and scales, it became a must to secure every project's supply chain of used packages and dependencies. Supply Chain Security is a rising concern in the current era and it's still under research and investigations. As part of making Jenkins-X an end-to-end solution for CI/CD I worked on both securing our own components and enabling our users to use these features during using our CI/CD for build and release steps.
```
Comment (babadofar) [inline content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md:16]: Perhaps shift focus to the security threats and that Jenkins X wants to show best practice implementation of Software Supply Chain security. 
Due to massive increase in Attacks on software supply chain, Jenkins X has to make efforts to ensure that the build process is secure. 
https://www.helpnetsecurity.com/2022/02/18/open-source-supply-chain/

**Review by babadofar — CHANGES_REQUESTED** [review]

File: content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md (hunk 1 above)
Comment (osamamagdy) [reply content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md]: Done

File: content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md (hunk 2 above)
Comment (osamamagdy) [reply content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md]: Done

File: content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md (hunk 3 above)
Comment (osamamagdy) [reply content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md]: Done

**Comment by babadofar** [comment]
The formatting is not looking good, you have to scroll sideways, text is capped inside a box, hard to read. 
Please fix this
image

**Comment by osamamagdy** [comment]
> The formatting is not looking good, you have to scroll sideways, text is capped inside a box, hard to read. Please fix this image

@babadofar  fixed

**Comment by babadofar** [comment]
Jenkins-x should be spelled Jenkins X. That is multiple times in the post

**Review by babadofar — APPROVED** [review]
Nice!

File: content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+The work for supply chain security is still in progress and there are many things to be done. At Jenkins X, we are planning to develop new features and integrate more solutions to provide a better supply chain security for our users. The list of things to be done are:
+
+* Add a cli subcommand in JX to verify sbom/signed binaries against the public keys.
+* Add [openSSF security scorecard](https://github.com/ossf/scorecard) to Jenkins X repositories.
+* Integrate [FRSCA](https://buildsec.github.io/frsca/) work with Jenkins X.
+
+## Acknowledgements
+
+Out of all internships and trainings I've done, this one was the most challenging, rewarding and educational experience. I would like to specially thank my mentor [Ankit Mohabatra](https://github.com/ankitm123) for his constant support and his patience to teach me a lot of his rich knowledge he gained by experience and thank the whole team of Jenkins X for their support and guidance and for welcoming me to the community. This internship was a great experience. I had a great time working on this project and I learned a lot from it. I hope to continue contributing to Jenkins X and the open-source community in the future. Thank you.
```
Comment (ankitm123) [inline content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md:97]: > Ankit Mohabatra

Ankit Mohapatra :grimacing:

File: content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+### Signing Jenkins X artifacts
+
+Description:
+
+Jenkins X is a collection of many components and tools that are used to provide the end-to-end CI/CD solution. Those components are used by our users and are also used by Jenkins X itself. It is important to sign those components to ensure the integrity of the components and the supply chain security of Jenkins X.
+
+Implementation:
+
+There are a set of tools developed by [sigstore](https://sigstore.dev/) to sign and verify artifacts. The most used tool in my implementation is [cosign](https://github.com/sigstore/cosign). I first started with signing the Jenkins X CLI with keys stored in github secrets and then moved to signing the Jenkins X docker images. After that, I switched to use cosign [keyless signing](https://github.com/sigstore/cosign/blob/main/KEYLESS.md) (an experimental feature using other Sigstore components to save the effort by key management ) The list of PRs for this work are:
```
Comment (ankitm123) [inline content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md:79]: > an experimental feature using other Sigstore components to save the effort by key management

an experimental feature using other Sigstore components (rekor and fulcio) to save end users from the hassles of key (mis)management.

**Review by ankitm123 — COMMENTED** [review]
Prelim comments.

Need to add the versioning work as well (printing version for `jx version`).

File: content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  Going through my GSoC journey with Jenkins X, this works as a documentation for our work to improve supply chain security of Jenkins X
+categories: [blog]
+keywords: [Community, GSoC, 2022, supply chain security, Report]
+slug: "gsoc-2022"
+aliases: []
+author: Osama Magdy
+---
+## Project Description
+
+Supply chain security is a rising concern is the current software era. Securing the software supply chain encompasses vulnerability remediation and the implementation of controls throughout the software development process. Due to massive increase in Attacks on software supply chain and the diversity of its [types](https://slsa.dev/spec/v0.1/threats), Jenkins X has to make efforts to ensure that the build process is secure. As part of making Jenkins X an end-to-end solution for CI/CD I worked on both securing our own components and enabling our users to use these features during using our CI/CD for build and release steps.
```
[the remark this answers is not in the document]
Comment (osamamagdy) [reply content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md:16]: Done

[the remark this answers is not in the document]
Comment (osamamagdy) [reply content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md:16]: Done

[the remark this answers is not in the document]
Comment (osamamagdy) [reply content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md:16]: Done

[the remark this answers is not in the document]
Comment (osamamagdy) [reply content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md:16]: Done

[the remark this answers is not in the document]
Comment (osamamagdy) [reply content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md:16]: Done

File: content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+Implementation
+
+I used the helm chart developed by [Chainguard](https://www.chainguard.dev/) for integrating Chains with Jenkins X. To integrate the [chart](https://github.com/chainguard-dev/tekton-helm-charts/tree/main/charts/tekton-chains) and added support for it on [jx3-versions](https://github.com/jenkins-x/jx3-versions) to make installation of helm chart easy for our users. The list of PRs for this:
+
+| PR                                                  | Short Description                                                           |
+| --------------------------------------------------- | --------------------------------------------------------------------------- |
+| https://github.com/jenkins-x/jx3-versions/pull/3359 | Supporting Tekton Chains helm chart with jx3-versions                       |
+| https://github.com/jenkins-x/jx-docs/pull/3660      | Documentation on how to install and integrate Tekton Chains with Jenknins-X |
```
[the remark this answers is not in the document]
Comment (osamamagdy) [reply content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md:40]: Done

File: content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md (hunk 4 above)
Comment (osamamagdy) [reply content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md:97]: Done 😅

File: content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md (hunk 5 above)
Comment (osamamagdy) [reply content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md:79]: Done

**Comment by osamamagdy** [comment]
Fixed in another PR :
https://github.com/jenkins-x/jx-docs/pull/3681

**Pull request merged**